### PR TITLE
[style] back button in review page and style thank you page

### PR DESCRIPTION
**Pull request by ethan-tam33**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- styled + added a functional back button to transition from `MatchingReviewScreen.tsx` to `MatchingSelectScreen.tsx`
- styled the thank you page in `StageSubmitted.tsx` following Austin's design, including a functional button to navigate back to home page
- added a loading spinner to the submit button in `MatchingReviewScreen.tsx` (NOTE: the loading spinner doesn't appear right now as `advanceToStage` doesn't have any async operations at the moment. #111 will include the database update in  `advanceToStage`, which should take long enough for the loading spinner to appear)

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

back button (not showing whole screen for adoptee privacy):
Screenshot 2025-12-22 at 10 41 52 AM

thank you page:
Screenshot 2025-12-22 at 10 38 07 AM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- go through the entire application process to interact with the ranking review page and thank you page

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

File: components/application/stages/StageSubmitted.tsx [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+            <p className="text-center text-[14px] leading-normal text-gray-11">
+              Thank you for submitting your application. Please check your email
+              for next steps and updates.
+            </p>
+          </div>
+
+          <ButtonLink href="/" variant="primary" className="py-2">
+            Back to Home
+          </ButtonLink>
+        </div>
```
Comment (InayaY3) [inline components/application/stages/StageSubmitted.tsx]: issue with spacing here

Image

**Review by InayaY3 — APPROVED** [review]
minor styling changes needed, but otherwise looks good!

File: components/application/stages/StageSubmitted.tsx (hunk 1 above)
Comment (ethan-tam33) [reply components/application/stages/StageSubmitted.tsx]: should be fixed now!
